### PR TITLE
[9.0.1] Take over #6769

### PR DESCRIPTION
**Pull request by masterleinad**
This PR was needed to compile with ICC-16.

### CODE REVIEW

**Comment by tjhei** [comment]
clang seems to be unhappy with this change?

**Comment by bangerth** [comment]
Hm, is this not the exact same code we have on mainline? Or did we further modify it there?

**Comment by masterleinad** [comment]
Avoiding unnamed namespaces in header files also affects the other commits. Fix incoming.

**Comment by masterleinad** [comment]
This should work now. Unfortunately, there are a lot of indentation changes which hide what we are really doing in the second commit.

**Comment by drwells** [comment]
I would really prefer to use the original patch and just let `clang-format` be unhappy so that we don't have hundreds of lines of changes here.

**Comment by bangerth** [comment]
Yes, I agree. 9.0 happened before the switch to clang-format, so let's stick with the formatting we used at the time.

**Comment by masterleinad** [comment]
The problem is rather that the indentation in #6769 is already `clang-format` conforming while the indentation for 9.0 is stil `astyle` and (`astyle` ->) `clang-format` -> `astyle` doesn't give nice indentation.

Let me try to make the required changes manually.

**Comment by masterleinad** [comment]
Updated. That should look much better.

File: source/base/mpi.cc [hunk 1]
```diff
@@ -23,7 +23,7 @@
 #include <deal.II/lac/la_parallel_block_vector.h>
 #include <deal.II/base/multithread_info.h>
 
-#include <numeric> 
+#include <numeric>
```
Comment (bangerth) [inline source/base/mpi.cc]: Not necessary, but also doesn't hurt :-)

Comment (masterleinad) [reply source/base/mpi.cc]: I get that after every change to the branch after running `astyle`. We should just fix it in one of the PRs. I guess we don't need an extra PR for that.

**Comment by drwells** [comment]
I think the tester may be stuck: it has been 17 hours. Since the tester doesn't use the intel compiler anyway its results aren't that useful; I think we should go ahead and merge in an hour or so even if it is not done.

**Comment by drwells** [comment]
Nevermind, it finally finished :)

**Pull request merged**